src/briefcase/platforms/macOS/__init__.py

```diff
@@ -382,10 +386,10 @@ def add_options(self, parser):
         )
 
     def verify_tools(self):
-        if self.tools.host_os != "Darwin":
-            raise BriefcaseCommandError(
-                "Code signing and / or building a DMG requires running on macOS."
-            )
+        # since the cli tools must be verified first, explicitly check host
+        # OS here instead of letting super().verify_tools() verify the OS.
+        if self.tools.host_os not in self.supported_host_os:
+            raise BriefcaseCommandError(self.supported_host_os_reason)
```

Comment:
As above, I wonder if this indicates a need for a generic `verify_host()` check

Reply:
Implemented `verify_host()`.